src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1007,13 +1008,29 @@ private ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] st
         }
 
         /// <inheritdoc />
-        public Task<Dictionary<uint256, Transaction>> GetTransactionsByStatusAsync(CrossChainTransferStatus status, bool sort = false)
+        public ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] statuses, bool sort = false)
         {
-            return Task.Run(() =>
-            {
-                ICrossChainTransfer[] res = this.GetTransfersByStatus(new[] { status }, sort);
-                return res.Where(t => t.PartialTransaction != null).ToDictionary(t => t.DepositTransactionId, t => t.PartialTransaction);
-            });
+            return this.GetTransfersByStatus(statuses, sort, false);
```

Comment:
The original code did not pass a bool=false
` return this.GetTransfersByStatus(statuses, sort, false);`

Reply:
I agree, `GetTransfersByStatus` must pass true to perform validation.  It's fine for `QueryTransfersByStatus` to not perform validation though as its used for console purposes. Not sure we even need two methods if we can default validation to true. You only need the original unchanged `GetTransfersByStatus` for which you can pass false for console purposes.

Reply:
```suggestion
            return this.GetTransfersByStatus(statuses, sort, true);
```

Reply:
```suggestion
            return this.GetTransfersByStatus(statuses, sort, true);
```